sdk/resourcemanager/Azure.ResourceManager.Core/src/ArmClient.cs

```diff
@@ -97,7 +97,7 @@ public ArmClient(TokenCredential credential, ArmClientOptions options = default)
             DefaultSubscription = string.IsNullOrWhiteSpace(defaultSubscriptionId)
                 ? GetDefaultSubscription()
                 : GetSubscriptions().TryGet(defaultSubscriptionId);
-            ClientOptions.ApiVersions.SetProviderClient(credential, baseUri, DefaultSubscription.Id.SubscriptionId);
+            ClientOptions.ApiVersions.SetProviderClient(credential, baseUri, defaultSubscriptionId ?? DefaultSubscription.Id.SubscriptionId);
```

Comment:
Is this change NOOP as DefaultSubscription has been initialized above?

Reply:
its NOOP and enables us to mock the construction of ArmClient since the default subscription won't actually be made in that case.